tests/modules/withconstructor.py

```diff
@@ -0,0 +1,47 @@
+from typing import List, Tuple
+
+from tests.modules import withenum
+from tests import modules
+from tests.modules.withenum import TimeUnit
+import datetime
+
+class Coordinates:
+    def __init__(self, x: float, y: float):
+        self.x = x
+        self.y = y
+
+class Point:
+    PI: float = 3.14159
+    origin = Coordinates(0, 0)
+
+    @staticmethod
+    def from_values(x: int, y: str, u: float, z: List[int]) -> 'Point':
+        return Point(x, y, u, z)
+
+    def get_coordinates(self):
+        return self.x, self.y
+
+    def __init__(self, x: int, y: str, unit: str, u: float, z: List[int]):
+        self.coordinates: Coordinates = Coordinates(x, float(y))
```

Comment:
type annotations are necessary for `py2puml` to infer the type of the instance attributes while parsing the AST.

`py2puml` is able to infer the type of self.coordinates in the following 2 examples:

1. inferred from a typed parameter in the constructor signature:

```python
def __init__(self, coordinates: Coordinates):
  self.coordinates = coordinates
```

2. inferred from a typed local variable of the constructor body:

```python
def __init__(self, x, y):
  coordinates: Coordinates = Coordinates(x, y)
  self.coordinates = coordinates
```

Of course, it would be possible to infer the type in other cases like:
- assigning a primitive-typed value (`self.x = 0`, `self.unit = 'meter'`)
- assigning an instance of another class (`self.origin = Coordinates(0, 0)`)

But that would be for another release. In any case, I do not want to execute and evaluate code from the constructor. I may attempt to connect to a database, read or write some file, etc. This is my red line. And parsing AST is fun :stuck_out_tongue_closed_eyes:

Reply:
Thanks for the summary. I also think `py2pulm` should stick to parsing ASTs.